jina/serve/runtimes/worker/http_fastapi_app.py

```diff
@@ -222,3 +233,62 @@ def _doc_to_event(doc):
         return {'event': 'update', 'data': doc.to_dict()}
     else:
         return {'event': 'update', 'data': doc.dict()}
+    
+def generate_gradio_interface(model: BaseModel):
```

Comment:
I think these have to be more ambitious and also cover docarray built-in types.

Reply:
I read that docarray is compatible with pydantic. Any thing that is different and you want me to pay special attention to?

Reply:
There are some special prebuilt types such as TextDoc, ImageDoc, etc ...